File: msgraph-mail/python/models/multi_value_legacy_extended_property.py
from __future__ import annotations
from kiota_abstractions.serialization import Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import entity

class MultiValueLegacyExtendedProperty(entity.Entity):
    def __init__(self,) -> None:
        """
        Instantiates a new MultiValueLegacyExtendedProperty and sets the default values.
        """
        super().__init__()
        # A collection of property values.
        self._value: Optional[List[str]] = None


    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> MultiValueLegacyExtendedProperty:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: MultiValueLegacyExtendedProperty
        """
        if not parse_node:
            raise Exception("parse_node cannot be undefined")
        return MultiValueLegacyExtendedProperty()

    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "value": lambda n : setattr(self, 'value', n.get_collection_of_primitive_values(str)),
        }
        super_fields = super().get_field_deserializers()
        fields.update(super_fields)
        return fields

    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if not writer:
            raise Exception("writer cannot be undefined")
        super().serialize(writer)
        writer.write_collection_of_primitive_values("value", self.value)

    @property
    def value(self,) -> Optional[List[str]]:
        """
        Gets the value property value. A collection of property values.
        Returns: Optional[List[str]]
        """
        return self._value

    @value.setter
    def value(self,value: Optional[List[str]] = None) -> None:
        """
        Sets the value property value. A collection of property values.
        Args:
            value: Value to set for the value property.
        """
        self._value = value

File: msgraph-mail/php/src/Models/MultiValueLegacyExtendedProperty.php
<?php

namespace Microsoft\Graph\Models;

use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

class MultiValueLegacyExtendedProperty extends Entity implements Parsable 
{
    /**
     * @var array<string>|null $value A collection of property values.
    */
    private ?array $value = null;
    
    /**
     * Instantiates a new MultiValueLegacyExtendedProperty and sets the default values.
    */
    public function __construct() {
        parent::__construct();
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return MultiValueLegacyExtendedProperty
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): MultiValueLegacyExtendedProperty {
        return new MultiValueLegacyExtendedProperty();
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return array_merge(parent::getFieldDeserializers(), [
            'value' => fn(ParseNode $n) => $o->setValue($n->getCollectionOfPrimitiveValues()),
        ]);
    }

    /**
     * Gets the value property value. A collection of property values.
     * @return array<string>|null
    */
    public function getValue(): ?array {
        return $this->value;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        parent::serialize($writer);
        $writer->writeCollectionOfPrimitiveValues('value', $this->value);
    }

    /**
     * Sets the value property value. A collection of property values.
     *  @param array<string>|null $value Value to set for the value property.
    */
    public function setValue(?array $value ): void {
        $this->value = $value;
    }

}
